actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -311,7 +311,7 @@ def session_options=(options)
     def GET
       @env["action_dispatch.request.query_parameters"] ||= Utils.deep_munge(normalize_encode_params(super || {}))
     rescue Rack::Utils::ParameterTypeError, Rack::Utils::InvalidParameterError => e
-      raise ActionController::BadRequest.new(:query, e)
+      raise ActionController::BadRequest, "Invalid query parameters: #{e.message}", e.backtrace
```

Comment:
Coming back to this… apps can raise `BadRequest.new` themselves, so this changes public API. Tricky.

We could raise an `ActionController::BadRequestParametersWithCause` instead that derives its message and backtrace from its `cause` exception:
```ruby
module ActionController
  class BadRequestParametersWithCause < BadRequest
    def initialize(type = nil)
      super type, cause
    end
  end
end
```